rostime/src/time.cpp

```diff
@@ -167,6 +167,29 @@ namespace ros
     nsec = nsec_sum;
 #endif
   }
+
+  void ros_monotonictime(uint32_t& sec, uint32_t& nsec)
+#ifndef WIN32
+    throw(NoHighPerformanceTimersException)
+#endif
+  {
+#ifndef WIN32
+#if HAS_CLOCK_GETTIME
+    timespec start;
+    clock_gettime(CLOCK_MONOTONIC, &start);
+    sec  = start.tv_sec;
+    nsec = start.tv_nsec;
+#else
+    // what to do if clock_gettime is not available???
+    struct timeval timeofday;
+    gettimeofday(&timeofday,NULL);
+    sec  = timeofday.tv_sec;
+    nsec = timeofday.tv_usec * 1000;
```

Comment:
Ok, it's been too long since I've looked at it to respond intelligently 😄, but I'll review the use of `CLOCK_MONOTONIC_RAW`. Thanks!